sklearn/utils/estimator_checks.py

```diff
@@ -1360,6 +1390,62 @@ def check_classifiers_train(name, classifier_orig):
                 assert_array_equal(np.argsort(y_log_prob), np.argsort(y_prob))
 
 
+def check_outliers_train(name, estimator_orig):
+    X, _ = make_blobs(n_samples=300, random_state=0)
+    X = shuffle(X, random_state=7)
+    n_samples, n_features = X.shape
+    estimator = clone(estimator_orig)
+    set_random_state(estimator)
+
+    # fit
+    estimator.fit(X)
+    # with lists
+    estimator.fit(X.tolist())
+
+    y_pred = estimator.predict(X)
+    assert_equal(y_pred.shape, (n_samples,))
+    # training set performance
+    # TODO
+
+    # raises error on malformed input for predict
+    assert_raises(ValueError, estimator.predict, X.T)
+
+    # decision_function agrees with predict
+    decision = estimator.decision_function(X)
+    assert_equal(decision.shape, (n_samples,))
+    dec_pred = (decision >= 0).astype(np.int)
+    dec_pred[dec_pred == 0] = -1
+    assert_array_equal(dec_pred, y_pred)
+
+    # raises error on malformed input for decision_function
+    assert_raises(ValueError, estimator.decision_function, X.T)
+
+    # decision_function is a translation of score_samples
+    y_scores = estimator.score_samples(X)
+    assert_equal(y_scores.shape, (n_samples,))
+    y_dec = y_scores - estimator.offset_
+    assert_array_equal(y_dec, decision)
+    assert_array_equal(np.argsort(y_scores), np.argsort(decision))
```

Comment:
You might want to use rankdata in case of ties.

Reply:
this is tested below with rankdata. Just remove it.